packages/atomic/src/components/common/facets/facet-common.tsx

```diff
@@ -253,6 +263,101 @@ export function facetShouldBeInitiallyCollapsed(
   return facetIndex + 1 > collapseFacetsAfter;
 }
 
+interface VisibilitySortedFacets {
+  visibleFacets: BaseFacetElement[];
+  invisibleFacets: BaseFacetElement[];
+}
+export function sortFacetVisibility(
+  facetElements: BaseFacetElement[],
+  facetInfoMap: FacetInfoMap
+): VisibilitySortedFacets {
+  const visibleFacets: BaseFacetElement[] = [];
+  const invisibleFacets: BaseFacetElement[] = [];
+
+  facetElements.forEach((facet) => {
+    if (facetInfoMap[facet.facetId] && facetInfoMap[facet.facetId].isHidden()) {
+      invisibleFacets.push(facet);
+    } else {
+      visibleFacets.push(facet);
+    }
+  });
+
+  return {visibleFacets, invisibleFacets};
+}
+
+export function updateCollapsedState(
+  facets: BaseFacetElement[],
+  collapseFacetsAfter: number
+) {
+  if (collapseFacetsAfter === -1) {
+    return;
+  }
+
+  facets.forEach((facet, index) => {
+    facet.isCollapsed = facetShouldBeInitiallyCollapsed(
+      index,
+      collapseFacetsAfter
+    );
+  });
+}
+export function updateCollapseFacetsAfter(
+  generator: AutomaticFacetGeneratorElement,
+  index: number,
+  collapseFacetsAfter: number
+) {
+  if (!generator) {
+    return;
+  }
+  const value = collapseFacetsAfter - index;
+  if (value > 0) {
+    generator.collapseFacetsAfter = value;
+  }
+  if (collapseFacetsAfter === -1) {
+    generator.collapseFacetsAfter = collapseFacetsAfter;
+  }
+}
+
+export function isTagNameAutomaticFacetGenerator(tagName: string) {
+  return tagName === 'ATOMIC-AUTOMATIC-FACET-GENERATOR';
+}
+
+function isPseudoFacet(el: Element): el is BaseFacetElement {
+  return 'facetId' in el;
+}
+
+export function getFacetsInChildren(
+  collection: HTMLCollection
+): BaseFacetElement[] {
+  const children = Array.from(collection);
+  const facets = children.filter((child) =>
+    isPseudoFacet(child)
+  ) as BaseFacetElement[];
+
+  return facets;
+}
+export function getGeneratorInChildren(
+  collection: HTMLCollection
+): AutomaticFacetGeneratorElement {
+  const children = Array.from(collection);
+
+  const automaticFacetGenerator = children.find((child) =>
+    isTagNameAutomaticFacetGenerator(child.tagName)
+  ) as AutomaticFacetGeneratorElement;
+
+  return automaticFacetGenerator;
+}
+
+export function sortFacetsViaManager(
```

Comment:
I would prefer the term `using`:
```suggestion
export function sortFacetsUsingManager(
```